pkg/catalog/manager/catalog_sync.go

```diff
@@ -28,9 +29,10 @@ func (m *Manager) Sync(key string, obj *v3.Catalog) error {
 
 	if commit == catalog.Status.Commit {
 		logrus.Debugf("Catalog %s is already up to date", catalog.Name)
-		if v3.CatalogConditionRefreshed.IsUnknown(catalog) {
+		if !v3.CatalogConditionRefreshed.IsTrue(catalog) {
```

Comment:
why do you need this change? The idea behind is that we set catalog to unknown when refreshing catalog, and if the commit is same and controller will just set condition to true and skip.

Reply:
if the commit is same but the credentials(e.g, username and password) are changed and the previous conditional status if false then we also need to update it once as edit catalog CRD will not set condition status to `unknown`.